File: serverless/serverless-node/src/handlers/putCryptoCurrency.js
const _ = require("lodash");
const { currencies } = require("../utils/currencies-list");
const currenciesApiService = require("../services/currencies-api/coindesk");
const currenciesService = require("../services/dynamodb/currencies");

module.exports.handler = async () => {
  const currencyParam = _.sample(currencies);
  try {
    const priceCurrency = await currenciesApiService
      .fetchPriceCurrency(currencyParam);
    const currencyCode = _.get(
      priceCurrency,
      `data.bpi[${currencyParam}].code`
    );
    const params = {
      currencyCode,
      priceCurrency
    };
    return await currenciesService.putCurrency(params);
  } catch (err) {
    return err;
  }
};

File: serverless/serverless-python/src/handlers/put_crypto_currency.py
import random
from ..services.currencies_api.coindesk import fetch_price_currency
from ..services.dynamodb.currencies import put_currency
from ..utils.currencies_list import currencies
from ..utils.object_helpers import parse_response_to_json


def handler(event, context):
  currency_param = random.choice(currencies)
  response = fetch_price_currency(currency_param)
  currency = parse_response_to_json(response)
  params = {
    'currency_code': currency_param,
    'currency': currency,
  }
  return put_currency(params)
